stellar_sdk/sep/stellar_web_authentication.py

```diff
@@ -233,6 +243,7 @@ def read_challenge_transaction(
         )
 
     # TODO: I don't think this is a good idea.
+    # What if we used the Java SDK's approach and had a ChallengeTransaction class?
```

Comment:
Yes, I would prefer to return a `class`, but returning a `tuple` seems to be pretty common in Python. In addition, the return value of this function is generally used within the SDK, so I think it's fine to use a tuple.

If you think it should be modified here to return a class, please let me know and we can handle it in another PR.

Reply:
Whether a `class` or `tuple` is returned is not too important, but returning a `class` would make returning additional data non-breaking compared to a `tuple`. Every time we wanted to return something new it would be a breaking change.

The downside to this implementation, which is non-breaking, is that the developer has to explicitly detect a client domain by iterating over the operations:

```
client_domain, client_domain_signing_key = None, None
challenge_tx = read_challenge_transaction(...)
for operation in challenge_tx.operations:
    if operation.data_name == "client_domain":
        client_domain = operation.data_value.decode()
        client_domain_signing_key = operation.source
```

If we had a `class`, we could add `client_domain` and `client_domain_signing_key` as attributes without breaking the API contract.

Reply:
Sounds great, let's change it in another independent PR.